docs/java/how-to-use-a-side-effect-in-java.md

```diff
@@ -0,0 +1,62 @@
+---
+id: how-to-use-a-side-effect-in-java
+title: How to use a Side Effect in Java
+sidebar_label: Side Effect
+description: A Side Effect is a method of execution to produce nondeterministic code.
+tags:
+  - Java
+  - developer-guide
+---
+
+To use a Side Effect in Java, set the [`sideEffect()`](<https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/workflow/Workflow.html#sideEffect(java.lang.Class,io.temporal.workflow.Functions.Func)>) function in your Workflow Execution and return the nondeterministic code.
+
+```java
+  int random = Workflow.sideEffect(Integer.class, () -> random.nextInt(100));
+  if random < 50 {
+         ....
+  } else {
+         ....
+  }
+```
+
+Here's another example that uses `sideEffect()`.
+
+```java
+// implementation of the @WorkflowMethod
+public void execute() {
+    int randomInt = Workflow.sideEffect( int.class, () -> {
+        Random random = new SecureRandom();
+        return random.nextInt();
+    });
+
+    String userHome = Workflow.sideEffect(String.class, () -> System.getenv("USER_HOME"));
+
+    if(randomInt % 2 == 0) {
+        // ...
+    } else {
+        // ...
+    }
+}
+```
+
+Java also provides a deterministic method to generate random numbers or random UUIDs.
```

Comment:
Same as above, looks slightly irrelevant that we teach basic Workflow.* methods here.

Something like "Temporal Workflow API provides ways to obtain most frequently used non-deterministic values like timestamps and random numbers in a deterministic way without using side-effects" and maybe a link to these methods or the descriptions of these APIs would be more appropriate here.